### PR TITLE
fix admin.access_log with filter startup fail 

### LINKED ISSUE #17210: Envoy crashes when using admin.access_log.filter in config
If we use admin.access_log.filter, Envoy immediately crashes on startup. I couldn't find a way to prevent the crash.
Please, see the config along with the startup log attached. 

Checked on **1.18.3**.

```yaml
admin:
  access_log:
  - name: envoy.access_loggers.stream
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog

    # !!! If we comment out the following two lines, the problem disappears
    filter:
      not_health_check_filter:  {}

```
Config: [envoy.yaml.txt](https://github.com/envoyproxy/envoy/files/6745992/envoy.yaml.txt)
Startup log: [envoy-crash.log](https://github.com/envoyproxy/envoy/files/6745987/envoy-crash.log)

### PR DESCRIPTION
**Pull request by chrisxrepo**
Signed-off-by: Chris Xin xinchuantao@qq.com

envoy.yaml set "admin.access_log.filter", then envoy startup fail

envoy.yaml
```
admin:
  access_log_path: "/tmp/envoy.log"
  address:
    socket_address:
      protocol: TCP
      address: 0.0.0.0
      port_value: 9901
  access_log:
  - name: envoy.access_loggers.stream
    typed_config:
      "@type": type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog
    filter:
      not_health_check_filter:  {}
```

startup log
```
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:338] initializing epoch 0 (base id=0, hot restart version=11.104)
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:340] statically linked extensions:
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.http.original_ip_detection: envoy.http.original_ip_detection.custom_header, envoy.http.original_ip_detection.xff
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.thrift_proxy.filters: envoy.filters.thrift.rate_limit, envoy.filters.thrift.router
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.clusters: envoy.cluster.eds, envoy.cluster.logical_dns, envoy.cluster.original_dst, envoy.cluster.static, envoy.cluster.strict_dns, envoy.clusters.aggregate, envoy.clusters.dynamic_forward_proxy, envoy.clusters.redis
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.http.stateful_header_formatters: preserve_case
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.http.cache: envoy.extensions.http.cache.simple
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.request_id: envoy.request_id.uuid
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.filters.http: envoy.bandwidth_limit, envoy.buffer, envoy.cors, envoy.csrf, envoy.ext_authz, envoy.ext_proc, envoy.fault, envoy.filters.http.adaptive_concurrency, envoy.filters.http.admission_control, envoy.filters.http.alternate_protocols_cache, envoy.filters.http.aws_lambda, envoy.filters.http.aws_request_signing, envoy.filters.http.bandwidth_limit, envoy.filters.http.buffer, envoy.filters.http.cache, envoy.filters.http.cdn_loop, envoy.filters.http.composite, envoy.filters.http.compressor, envoy.filters.http.cors, envoy.filters.http.csrf, envoy.filters.http.decompressor, envoy.filters.http.dynamic_forward_proxy, envoy.filters.http.dynamo, envoy.filters.http.ext_authz, envoy.filters.http.ext_proc, envoy.filters.http.fault, envoy.filters.http.grpc_http1_bridge, envoy.filters.http.grpc_http1_reverse_bridge, envoy.filters.http.grpc_json_transcoder, envoy.filters.http.grpc_stats, envoy.filters.http.grpc_web, envoy.filters.http.header_to_metadata, envoy.filters.http.health_check, envoy.filters.http.ip_tagging, envoy.filters.http.jwt_authn, envoy.filters.http.local_ratelimit, envoy.filters.http.lua, envoy.filters.http.oauth2, envoy.filters.http.on_demand, envoy.filters.http.original_src, envoy.filters.http.ratelimit, envoy.filters.http.rbac, envoy.filters.http.router, envoy.filters.http.set_metadata, envoy.filters.http.squash, envoy.filters.http.tap, envoy.filters.http.wasm, envoy.grpc_http1_bridge, envoy.grpc_json_transcoder, envoy.grpc_web, envoy.health_check, envoy.http_dynamo_filter, envoy.ip_tagging, envoy.local_rate_limit, envoy.lua, envoy.rate_limit, envoy.router, envoy.squash, match-wrapper
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.resolvers: envoy.ip
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.matching.http.input: request-headers, request-trailers, response-headers, response-trailers
[2021-07-12 03:46:47.088][33078][info][main] [source/server/server.cc:342]   envoy.health_checkers: envoy.health_checkers.redis
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.matching.action: composite-action, skip
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.compression.decompressor: envoy.compression.brotli.decompressor, envoy.compression.gzip.decompressor
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.retry_priorities: envoy.retry_priorities.previous_priorities
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.filters.listener: envoy.filters.listener.http_inspector, envoy.filters.listener.original_dst, envoy.filters.listener.original_src, envoy.filters.listener.proxy_protocol, envoy.filters.listener.tls_inspector, envoy.listener.http_inspector, envoy.listener.original_dst, envoy.listener.original_src, envoy.listener.proxy_protocol, envoy.listener.tls_inspector
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.grpc_credentials: envoy.grpc_credentials.aws_iam, envoy.grpc_credentials.default, envoy.grpc_credentials.file_based_metadata
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.filters.udp_listener: envoy.filters.udp.dns_filter, envoy.filters.udp_listener.udp_proxy
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.quic.proof_source: envoy.quic.proof_source.filter_chain
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.transport_sockets.downstream: envoy.transport_sockets.alts, envoy.transport_sockets.quic, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.starttls, envoy.transport_sockets.tap, envoy.transport_sockets.tls, raw_buffer, starttls, tls
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.formatter: envoy.formatter.req_without_query
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.rate_limit_descriptors: envoy.rate_limit_descriptors.expr
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.bootstrap: envoy.bootstrap.wasm, envoy.extensions.network.socket_interface.default_socket_interface
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.internal_redirect_predicates: envoy.internal_redirect_predicates.allow_listed_routes, envoy.internal_redirect_predicates.previous_routes, envoy.internal_redirect_predicates.safe_cross_scheme
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.access_loggers: envoy.access_loggers.file, envoy.access_loggers.http_grpc, envoy.access_loggers.open_telemetry, envoy.access_loggers.stderr, envoy.access_loggers.stdout, envoy.access_loggers.tcp_grpc, envoy.access_loggers.wasm, envoy.file_access_log, envoy.http_grpc_access_log, envoy.open_telemetry_access_log, envoy.stderr_access_log, envoy.stdout_access_log, envoy.tcp_grpc_access_log, envoy.wasm_access_log
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.thrift_proxy.transports: auto, framed, header, unframed
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.dubbo_proxy.protocols: dubbo
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.thrift_proxy.protocols: auto, binary, binary/non-strict, compact, twitter
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.filters.network: envoy.client_ssl_auth, envoy.echo, envoy.ext_authz, envoy.filters.network.client_ssl_auth, envoy.filters.network.connection_limit, envoy.filters.network.direct_response, envoy.filters.network.dubbo_proxy, envoy.filters.network.echo, envoy.filters.network.ext_authz, envoy.filters.network.http_connection_manager, envoy.filters.network.kafka_broker, envoy.filters.network.local_ratelimit, envoy.filters.network.mongo_proxy, envoy.filters.network.mysql_proxy, envoy.filters.network.postgres_proxy, envoy.filters.network.ratelimit, envoy.filters.network.rbac, envoy.filters.network.redis_proxy, envoy.filters.network.rocketmq_proxy, envoy.filters.network.sni_cluster, envoy.filters.network.sni_dynamic_forward_proxy, envoy.filters.network.tcp_proxy, envoy.filters.network.thrift_proxy, envoy.filters.network.wasm, envoy.filters.network.zookeeper_proxy, envoy.http_connection_manager, envoy.mongo_proxy, envoy.ratelimit, envoy.redis_proxy, envoy.tcp_proxy
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.matching.common_inputs: envoy.matching.common_inputs.environment_variable
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.upstream_options: envoy.extensions.upstreams.http.v3.HttpProtocolOptions, envoy.upstreams.http.http_protocol_options
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.retry_host_predicates: envoy.retry_host_predicates.omit_canary_hosts, envoy.retry_host_predicates.omit_host_metadata, envoy.retry_host_predicates.previous_hosts
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.compression.compressor: envoy.compression.brotli.compressor, envoy.compression.gzip.compressor
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.quic.server.crypto_stream: envoy.quic.crypto_stream.server.quiche
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.dubbo_proxy.route_matchers: default
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.upstreams: envoy.filters.connection_pools.tcp.generic
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.dubbo_proxy.filters: envoy.filters.dubbo.router
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.resource_monitors: envoy.resource_monitors.fixed_heap, envoy.resource_monitors.injected_resource
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.matching.input_matchers: envoy.matching.matchers.consistent_hashing, envoy.matching.matchers.ip
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.tracers: envoy.dynamic.ot, envoy.lightstep, envoy.tracers.datadog, envoy.tracers.dynamic_ot, envoy.tracers.lightstep, envoy.tracers.opencensus, envoy.tracers.skywalking, envoy.tracers.xray, envoy.tracers.zipkin, envoy.zipkin
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.transport_sockets.upstream: envoy.transport_sockets.alts, envoy.transport_sockets.quic, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.starttls, envoy.transport_sockets.tap, envoy.transport_sockets.tls, envoy.transport_sockets.upstream_proxy_protocol, raw_buffer, starttls, tls
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.wasm.runtime: envoy.wasm.runtime.null, envoy.wasm.runtime.v8
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.tls.cert_validator: envoy.tls.cert_validator.default, envoy.tls.cert_validator.spiffe
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.dubbo_proxy.serializers: dubbo.hessian2
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.stats_sinks: envoy.dog_statsd, envoy.graphite_statsd, envoy.metrics_service, envoy.stat_sinks.dog_statsd, envoy.stat_sinks.graphite_statsd, envoy.stat_sinks.hystrix, envoy.stat_sinks.metrics_service, envoy.stat_sinks.statsd, envoy.stat_sinks.wasm, envoy.statsd
[2021-07-12 03:46:47.089][33078][info][main] [source/server/server.cc:342]   envoy.guarddog_actions: envoy.watchdog.abort_action, envoy.watchdog.profile_action
[2021-07-12 03:46:47.111][33078][warning][misc] [source/common/protobuf/message_validator_impl.cc:21] Deprecated field: type envoy.config.bootstrap.v3.Admin Using deprecated option 'envoy.config.bootstrap.v3.Admin.access_log_path' from file bootstrap.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/version_history/version_history for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.
[2021-07-12 03:46:47.111][33078][info][main] [source/server/server.cc:358] HTTP header map info:
[2021-07-12 03:46:47.116][33078][info][main] [source/server/server.cc:361]   request header map: 632 bytes: :authority,:method,:path,:protocol,:scheme,accept,accept-encoding,access-control-request-method,authentication,authorization,cache-control,cdn-loop,connection,content-encoding,content-length,content-type,expect,grpc-accept-encoding,grpc-timeout,if-match,if-modified-since,if-none-match,if-range,if-unmodified-since,keep-alive,origin,pragma,proxy-connection,referer,te,transfer-encoding,upgrade,user-agent,via,x-client-trace-id,x-envoy-attempt-count,x-envoy-decorator-operation,x-envoy-downstream-service-cluster,x-envoy-downstream-service-node,x-envoy-expected-rq-timeout-ms,x-envoy-external-address,x-envoy-force-trace,x-envoy-hedge-on-per-try-timeout,x-envoy-internal,x-envoy-ip-tags,x-envoy-max-retries,x-envoy-original-path,x-envoy-original-url,x-envoy-retriable-header-names,x-envoy-retriable-status-codes,x-envoy-retry-grpc-on,x-envoy-retry-on,x-envoy-upstream-alt-stat-name,x-envoy-upstream-rq-per-try-timeout-ms,x-envoy-upstream-rq-timeout-alt-response,x-envoy-upstream-rq-timeout-ms,x-forwarded-client-cert,x-forwarded-for,x-forwarded-proto,x-ot-span-context,x-request-id
[2021-07-12 03:46:47.116][33078][info][main] [source/server/server.cc:361]   request trailer map: 136 bytes:
[2021-07-12 03:46:47.116][33078][info][main] [source/server/server.cc:361]   response header map: 432 bytes: :status,access-control-allow-credentials,access-control-allow-headers,access-control-allow-methods,access-control-allow-origin,access-control-expose-headers,access-control-max-age,age,cache-control,connection,content-encoding,content-length,content-type,date,etag,expires,grpc-message,grpc-status,keep-alive,last-modified,location,proxy-connection,server,transfer-encoding,upgrade,vary,via,x-envoy-attempt-count,x-envoy-decorator-operation,x-envoy-degraded,x-envoy-immediate-health-check-fail,x-envoy-ratelimited,x-envoy-upstream-canary,x-envoy-upstream-healthchecked-cluster,x-envoy-upstream-service-time,x-request-id
[2021-07-12 03:46:47.116][33078][info][main] [source/server/server.cc:361]   response trailer map: 160 bytes: grpc-message,grpc-status
[2021-07-12 03:46:47.123][33078][critical][assert] [./source/common/singleton/threadsafe_singleton.h:56] assert failure: loader_ != nullptr. Details: InjectableSingleton used prior to initialization
[2021-07-12 03:46:47.123][33078][critical][backtrace] [./source/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x8136
[2021-07-12 03:46:47.123][33078][critical][backtrace] [./source/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
[2021-07-12 03:46:47.123][33078][critical][backtrace] [./source/server/backtrace.h:92] Envoy version: 7c1991ffe382bacf2bf415a643c5ea7428d0880e/1.19.0-dev/Modified/DEBUG/BoringSSL
[2021-07-12 03:46:47.168][33078][critical][backtrace] [./source/server/backtrace.h:96] #0: Envoy::SignalAction::sigHandler() [0x55913cd2f2ac]
[2021-07-12 03:46:47.168][33078][critical][backtrace] [./source/server/backtrace.h:96] #1: __restore_rt [0x7f996f243980]
[2021-07-12 03:46:47.212][33078][critical][backtrace] [./source/server/backtrace.h:96] #2: Envoy::Server::InstanceImpl::runtime() [0x55913b6dbca1]
[2021-07-12 03:46:47.257][33078][critical][backtrace] [./source/server/backtrace.h:96] #3: Envoy::Server::ServerFactoryContextImpl::runtime() [0x55913b69a210]
[2021-07-12 03:46:47.301][33078][critical][backtrace] [./source/server/backtrace.h:96] #4: Envoy::AccessLog::AccessLogFactory::fromProto() [0x55913c53e4a1]
[2021-07-12 03:46:47.346][33078][critical][backtrace] [./source/server/backtrace.h:96] #5: Envoy::Server::Configuration::InitialImpl::InitialImpl() [0x55913c0ea8df]
[2021-07-12 03:46:47.390][33078][critical][backtrace] [./source/server/backtrace.h:96] #6: Envoy::Server::InstanceImpl::initialize() [0x55913b6d3b4b]
[2021-07-12 03:46:47.433][33078][critical][backtrace] [./source/server/backtrace.h:96] #7: Envoy::Server::InstanceImpl::InstanceImpl() [0x55913b6cf8da]
[2021-07-12 03:46:47.478][33078][critical][backtrace] [./source/server/backtrace.h:96] #8: std::__1::make_unique<>() [0x55913708d703]
[2021-07-12 03:46:47.522][33078][critical][backtrace] [./source/server/backtrace.h:96] #9: Envoy::MainCommonBase::MainCommonBase() [0x559137085790]
[2021-07-12 03:46:47.566][33078][critical][backtrace] [./source/server/backtrace.h:96] #10: Envoy::MainCommon::MainCommon() [0x559137087db1]
[2021-07-12 03:46:47.610][33078][critical][backtrace] [./source/server/backtrace.h:96] #11: std::__1::make_unique<>() [0x55913708ec64]
[2021-07-12 03:46:47.654][33078][critical][backtrace] [./source/server/backtrace.h:96] #12: Envoy::MainCommon::main() [0x559137088402]
[2021-07-12 03:46:47.697][33078][critical][backtrace] [./source/server/backtrace.h:96] #13: main [0x559137038c8b]
[2021-07-12 03:46:47.697][33078][critical][backtrace] [./source/server/backtrace.h:96] #14: __libc_start_main [0x7f996ee61bf7]
Aborted (core dumped)
```

crash stack from core dump file
```
(gdb) bt
#0  0x00007f3ef5134817 in raise () from /lib/x86_64-linux-gnu/libpthread.so.0
#1  0x0000559b56c2259f in Envoy::SignalAction::sigHandler (sig=6, info=0x7f3ef5d12bb0, context=0x7f3ef5d12a80) at source/common/signal/signal_action.cc:53
#2  <signal handler called>
#3  0x00007f3ef4d6ffb7 in raise () from /lib/x86_64-linux-gnu/libc.so.6
#4  0x00007f3ef4d71921 in abort () from /lib/x86_64-linux-gnu/libc.so.6
#5  0x0000559b5558e615 in Envoy::InjectableSingleton<Envoy::Runtime::Loader>::get () at ./source/common/singleton/threadsafe_singleton.h:56
#6  0x0000559b555ceca1 in Envoy::Server::InstanceImpl::runtime (this=0x55323fc57300) at source/server/server.cc:860
#7  0x0000559b5558d210 in Envoy::Server::ServerFactoryContextImpl::runtime (this=0x55323fc58b20) at ./source/server/server.h:182
#8  0x0000559b564314a1 in Envoy::AccessLog::AccessLogFactory::fromProto (config=..., context=...) at source/common/access_log/access_log_impl.cc:310
#9  0x0000559b55fdd8df in Envoy::Server::Configuration::InitialImpl::InitialImpl (this=0x7ffd35604640, bootstrap=..., options=..., server=...) at source/server/configuration_impl.cc:202
#10 0x0000559b555c6b4b in Envoy::Server::InstanceImpl::initialize (this=0x55323fc57300, options=..., local_address=..., component_factory=..., hooks=...) at source/server/server.cc:452
#11 0x0000559b555c28da in Envoy::Server::InstanceImpl::InstanceImpl (this=0x55323fc57300, init_manager=..., options=..., time_system=..., local_address=..., hooks=..., restarter=..., store=..., access_log_lock=..., component_factory=..., random_generator=..., tls=...,
    thread_factory=..., file_system=..., process_context=..., watermark_factory=...) at source/server/server.cc:107
#12 0x0000559b50f80703 in std::__1::make_unique<Envoy::Server::InstanceImpl, Envoy::Init::Manager&, Envoy::Server::Options const&, Envoy::Event::TimeSystem&, std::__1::shared_ptr<Envoy::Network::Address::Instance const>&, Envoy::ListenerHooks&, Envoy::Server::HotRestart&, Envoy::Stats::ThreadLocalStoreImpl&, Envoy::Thread::BasicLockable&, Envoy::Server::ComponentFactory&, std::__1::unique_ptr<Envoy::Random::RandomGenerator, std::__1::default_delete<Envoy::Random::RandomGenerator> >, Envoy::ThreadLocal::InstanceImpl&, Envoy::Thread::ThreadFactory&, Envoy::Filesystem::Instance&, std::__1::unique_ptr<Envoy::ProcessContext, std::__1::default_delete<Envoy::ProcessContext> > > (__args=..., __args=..., __args=..., __args=..., __args=..., __args=..., __args=..., __args=..., __args=..., __args=..., __args=...,
    __args=..., __args=..., __args=...) at /opt/llvm/bin/../include/c++/v1/memory:2927
#13 0x0000559b50f78790 in Envoy::MainCommonBase::MainCommonBase (this=0x55323fd02fb8, options=..., time_system=..., listener_hooks=..., component_factory=..., platform_impl=..., random_generator=..., process_context=...) at source/exe/main_common.cc:88
#14 0x0000559b50f7adb1 in Envoy::MainCommon::MainCommon (this=0x55323fd02900, argc=3, argv=0x7ffd356066b8) at source/exe/main_common.cc:207
#15 0x0000559b50f81c64 in std::__1::make_unique<Envoy::MainCommon, int&, char**&> (__args=@0x7ffd35606530: 0x7ffd356066b8, __args=@0x7ffd35606530: 0x7ffd356066b8) at /opt/llvm/bin/../include/c++/v1/memory:2927
#16 0x0000559b50f7b402 in Envoy::MainCommon::main(int, char**, std::__1::function<void (Envoy::Server::Instance&)>) (argc=3, argv=0x7ffd356066b8, hook=...) at source/exe/main_common.cc:234
#17 0x0000559b50f2bc8b in main (argc=3, argv=0x7ffd356066b8) at source/exe/main.cc:24
```

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #17210]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by dmitri-d** [comment]
This PR replaces https://github.com/envoyproxy/envoy/pull/17235/

File: test/server/configuration_impl_test.cc [hunk 1]
```diff
@@ -1040,12 +1042,48 @@ TEST_F(ConfigurationImplTest, DEPRECATED_FEATURE_TEST(DeprecatedAccessLogPathWit
...
   NiceMock<Server::MockInstance> server;
-  InitialImpl config(bootstrap, options, server_);
+  InitialImpl config(bootstrap, options);
+  config.initAccessLog(bootstrap, server_);
   Network::MockListenSocket socket_mock;
 
   ASSERT_EQ(config.admin().accessLogs().size(), 2);
 }
 
+TEST_F(ConfigurationImplTest, DEPRECATED_FEATURE_TEST(AccessLogWithFilter)) {
```
Comment (dmitri-d) [inline test/server/configuration_impl_test.cc:1052]: Not sure why you are using `DEPRECATED_FEATURE_TEST` here; only `access_log_path` field has been deprecated, which you are not using in the test.

Comment (dmitri-d) [reply test/server/configuration_impl_test.cc:1052]: Could you also add a test to verify that `access_log_path` is handled correctly? You'll need to use `DEPRECATED_FEATURE_TEST` for it.

Comment (dmitri-d) [reply test/server/configuration_impl_test.cc:1052]: And also a test that verifies that an empty admin config is ok too.

**Comment by dmitri-d** [comment]
lgtm, provided CI is happy. @chrisxrepo: please don't force-push changes in the future -- it makes it harder to review PRs.

**Comment by dmitri-d** [comment]
@envoyproxy/senior-maintainers for final pass please

File: source/server/server.cc [hunk 2]
```diff
@@ -553,8 +557,6 @@ void InstanceImpl::initialize(const Options& options,
 
   // Runtime gets initialized before the main configuration since during main configuration
   // load things may grab a reference to the loader for later use.
-  runtime_singleton_ = std::make_unique<Runtime::ScopedLoaderSingleton>(
-      component_factory.createRuntime(*this, initial_config));
   hooks.onRuntimeCreated();
```
Comment (mattklein123) [inline source/server/server.cc:560]: This comment and the hook should also get moved up.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks for the fix. One comment, but can you also provide the full crash stack? Is the issue that the admin log code is using runtime before it's available? I think this fix is OK but another fix would be to just check to see if runtime exists before using it at the crashing call site (depending on where it is.)

/wait

File: source/server/configuration_impl.cc [hunk 3]
```diff
@@ -245,6 +231,25 @@ InitialImpl::InitialImpl(const envoy::config::bootstrap::v3::Bootstrap& bootstra
   }
 }
 
+void InitialImpl::initAccessLog(const envoy::config::bootstrap::v3::Bootstrap& bootstrap,
```
Comment (mattklein123) [inline source/server/configuration_impl.cc:234]: Can we call this `initAdminAccessLog()`?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks LGTM with small comment.

/wait

File: source/server/configuration_impl.cc (hunk 3 above)
Comment (chrisxrepo) [reply source/server/configuration_impl.cc:234]: ok

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**